sympy/core/numbers.py

```diff
@@ -3972,17 +3972,22 @@ def _eval_power(self, expt):
         I**3 mod 4 -> -I
         """
 
-        if isinstance(expt, Number):
-            if isinstance(expt, Integer):
-                expt = expt.p % 4
-                if expt == 0:
-                    return S.One
-                if expt == 1:
-                    return S.ImaginaryUnit
-                if expt == 2:
-                    return -S.One
+        if isinstance(expt, Integer):
+            expt = expt % 4
+            if expt == 0:
+                return S.One
+            elif expt == 1:
+                return S.ImaginaryUnit
+            elif expt == 2:
+                return S.NegativeOne
+            elif expt == 3:
                 return -S.ImaginaryUnit
-        return
+        if isinstance(expt, Rational):
+            i, r = divmod(expt, 2)
+            rv = Pow(S.ImaginaryUnit, r, evaluate=False)
+            if i:
```

Comment:
This should probably be `if i % 2`:
```
>>> I**Rational(5, 2)
-sqrt(I)
>>> I**Rational(9, 2)
-sqrt(I)
```
`sqrt(I)**n` will not be evaluated by this code.

Reply:
Thanks -- I kept switching between Mod and divmod thinking.

Hmm... so `sqrt(I)**5 -> I**(5/2)` because it is a Pow operation and sqrt(I) is the base so it won't pass through this code. Then it should also be handled in the Pow instantiation by looking for a base that is a rational power of `I` and an exponent that is Rational.

Reply:
This change has been made. Any other changes? Else I think this is ready and is a pretty low-impact change.

Reply:
I will confirm, but I don't think special handling in Pow needs to happen since the `_eval_power` should handle it -- it is going to be a lot slower there, however, I think and maybe is worth the quick handling at the outset.

Reply:
Unless you think better to do so, I'll let the current Pow code handle `sqrt(I)**n` where `n` is a Rational.